src/service/invoice-service.ts

```diff
@@ -205,27 +186,23 @@ export default class InvoiceService {
    * Creates a Transfer for an Invoice from TransactionResponses
    * @param forId - The user which receives the Invoice/Transfer
    * @param transactions - The array of transactions which to create the Transfer for
+   * @param amount - The amount to transfer
    */
-  public async createTransferFromTransactions(forId: number,
-    transactions: Transaction[]): Promise<TransferResponse> {
+  public async createTransfer(forId: number,
+    transactions: Transaction[], amount: DineroObjectRequest): Promise<TransferResponse> {
     const transactionId = transactions.length > 0 ? transactions.map((t) => t.id) : null;
     const baseTransactions = (await new TransactionService(this.manager).getTransactions({ transactionId })).records;
 
-    const dineroObjectRequest: DineroObjectRequest = {
-      amount: 0,
-      currency: dinero.defaultCurrency,
-      precision: dinero.defaultPrecision,
-    };
+    const sum = baseTransactions.reduce((acc, t) => {
+      if (t.from.id !== forId) throw new Error(`Transaction from ${t.from.id} not from user ${forId}`);
+      return acc + t.value.amount;
+    }, 0);
 
-    if (baseTransactions.length !== 0) {
-      baseTransactions.forEach((t) => {
-        if (t.from.id !== forId) throw new Error(`Transaction from ${t.from.id} not from user ${forId}`);
-        dineroObjectRequest.amount += t.value.amount;
-      });
-    }
+    // TODO: Should we error if the sum is not equal to the amount?
```

Comment:
No, I don't think we should. Like with the `SellerPayout`, I think it is best to just make it very clear in the frontend that something does not add up. Then, you can fix it afterwards. Creating invoices is a reversable action, right?